### PR TITLE
Migrated authentication application from FBV to CBV

### PR DESCRIPTION
**Pull request by amruthasangeeth**
Authentication app used function based views for login and logout process. It is now migrated to CBV using Django's built in auth views.

Related to #288.

### CODE REVIEW

**Comment by tapasweni-pathak** [comment]
@amruthasangeeth In every pull request related to this change, please reference the issue #288. I have done this for you in the last two pull requests.

File: vms/vms/views.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+from django.views.generic.edit import FormView
+
+from django.core.mail import send_mail
+from django.core.mail.message import BadHeaderError
+from django.http.response import HttpResponse
+
+from vms import settings
+
+def anonymous_required(func):
+    def as_view(request, *args, **kwargs):
```
Comment (tapasweni-pathak) [inline vms/vms/views.py]: @amruthasangeeth Why this name? I am not getting the context. :(

File: vms/vms/urls.py [hunk 2]
```diff
@@ -16,4 +21,13 @@
     url(r'^registration/', include('registration.urls', namespace='registration')),
     url(r'^shift/', include('shift.urls', namespace='shift')),
     url(r'^volunteer/', include('volunteer.urls', namespace="volunteer")),
+    url(r'^portal', TemplateView.as_view(template_name='home/home.html'),name='home'),
+    url(r'^login/$',
+         anonymous_required(auth_views.login),
+        {'template_name': 'authentication/login.html'},
```
Comment (tapasweni-pathak) [inline vms/vms/urls.py]: @amruthasangeeth Why all this is not in a single line? Has pep8 done this?

File: vms/vms/views.py (hunk 1 above)
Comment (amruthasangeeth) [inline vms/vms/views.py]: @tapasweni-pathak The mixin that we used is AnonymousRequiredMixin. So thought of giving this name for better understanding

File: vms/vms/urls.py (hunk 2 above)
Comment (amruthasangeeth) [inline vms/vms/urls.py]: I see that the indentation is given as both single line and multi line. When it comes to password reset and email confirmation we can follow the same indentation.

Comment (tapasweni-pathak) [inline vms/vms/urls.py]: @amruthasangeeth If we put it like
```
url(r'^login/$', anonymous_required(auth_views.login), {'template_name': 'authentication/login.html'}, name='login_process'),
 
url(r'^user/logout/$', auth_views.logout, {'template_name': 'home/home.html'}, name='logout_process'),

```
are we violating any pep8 rules?

Comment (amruthasangeeth) [inline vms/vms/urls.py]: No, It is not.

File: vms/vms/views.py (hunk 1 above)
Comment (tapasweni-pathak) [inline vms/vms/views.py]: @amruthasangeeth Okay. 

What is this part of code doing for us? This is for the authentication needed for login/logout, right? Can we name it something which is close to the work this is doing for us. 

After this I also see a need of having comments on top of the functions in the view files. Can you please add a docstring on top of this function. We will start from this. I'll check if there are many, in that case I'll open an issue.

I also want to discuss a bit about this mixin. Ping me when you come on Slack, I'm available till 4:55 P.M.

File: vms/vms/urls.py (hunk 2 above)
Comment (tapasweni-pathak) [inline vms/vms/urls.py]: @amruthasangeeth Okay. Cool. Then please do this change. It is more readable and closely related to the indentation others are following.

**Comment by tapasweni-pathak** [comment]
@amruthasangeeth Are you planning to update this one or will you submit a new pull request for this change?

**Comment by amruthasangeeth** [comment]
@tapasweni-pathak I am planning to update it along with next pull request. Is that fine?

**Comment by tapasweni-pathak** [comment]
@amruthasangeeth That will be great. Please submit the next part of work as separate pull request.

**Pull request closed without merging**